### PR TITLE
[ftr] Added a feature of exporting selected and all groups' EICs as pdf

### PR DESCRIPTION
**Pull request by sakshikukreja14**
Earlier only the selected group of EICs could be exported as pdf. A new functionality has been added where in, user can export all the EICs as pdf and also set of selected EICs can be exported.

### CODE REVIEW

File: src/gui/mzroll/eicwidget.cpp [hunk 1]
```diff
@@ -2288,3 +2288,422 @@ void EicWidget::addMS2Events(float mzmin, float mzmax)
...
+    ////Barplot////
+    auto barplot = new BarPlot(NULL, 0);
+    scene.addItem(barplot);
+
+    barplot->setMainWindow(getMainWindow());
+    barplot->setPeakGroup(group);
+
+    int bwidth = barplot->boundingRect().width();
+    int bheight = barplot->boundingRect().height();
+    int xpos = scene.width() * 0.95 - 130;
```
Comment (saifulbkhan) [inline src/gui/mzroll/eicwidget.cpp:2699]: I think this hardcoded `- 130` offset does not work for all devices. For instance, I am getting this kind of an image:
Screenshot 2020-05-06 at 3 10 48 PM
The bars for the highest intensity peaks got clipped out.

Use a dynamic offset, maybe using the barplot item's overall width itself.

File: src/gui/mzroll/tabledockwidget.h [hunk 2]
```diff
@@ -203,6 +203,8 @@ public Q_SLOTS:
   void saveModel();
 
   void printPdfReport();
+  void renderPdf(QString fileName);
+  void pdfReadyNotification();
```
Comment (saifulbkhan) [inline src/gui/mzroll/tabledockwidget.h:207]: Add some docstrings, whenever you are adding new functions/variables.

File: src/gui/mzroll/eicwidget.cpp [hunk 3]
```diff
@@ -2288,3 +2288,422 @@ void EicWidget::addMS2Events(float mzmin, float mzmax)
...
+
+    for (int i = 0; i < eicparameters->peakgroups.size(); i++) {
+        if (mzUtils::checkOverlap(eicparameters->peakgroups[i].minRt,
+                                  eicparameters->peakgroups[i].maxRt, eicParameters->_slice.rtmin,
+                                  eicparameters->_slice.rtmax) > 0) {
+            if (eicparameters->peakgroups[i].maxHeightIntensity > potentialMaxY) {
+                potentialMaxY = eicparameters->peakgroups[i].maxHeightIntensity;
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (sakshikukreja14) [reply src/gui/mzroll/eicwidget.cpp:2463]: Yeaa.. I did that earlier but in visualisation of EIC this logic is used. If I used 1.2 times, there are chances that if pdf the groups that are made have lesser height which leads to inconsistency of shape in pdf and in EIC Widget

File: src/gui/mzroll/eicwidget.cpp [hunk 4]
```diff
@@ -2288,3 +2288,422 @@ void EicWidget::addMS2Events(float mzmin, float mzmax)
...
+                              getMainWindow()->mavenParameters->minQuality,
+                              getMainWindow()->mavenParameters->distXWeight,
+                              getMainWindow()->mavenParameters->distYWeight,
+                              getMainWindow()->mavenParameters->overlapWeight,
+                              getMainWindow()->mavenParameters->useOverlap,
+                              getMainWindow()->mavenParameters->minSignalBaselineDifference,
+                              getMainWindow()->mavenParameters->fragmentTolerance,
+                              getMainWindow()->mavenParameters->scoringAlgo);
+
+
```
[the remark this answers is not in the document]
Comment (sakshikukreja14) [reply src/gui/mzroll/eicwidget.cpp:2440]: umm.. this was also a part for getting correct y bound for axis

File: src/gui/mzroll/eicwidget.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (saifulbkhan) [reply src/gui/mzroll/eicwidget.cpp:2312]: I think that is fine. Shape and height should be set according to the peak-group in consideration - both in the PDF and the EIC widget. You need not change the EIC widget logic for now, but let's make that the case for PDF exports at least.

[the remark this answers is not in the document]
Comment (sakshikukreja14) [reply src/gui/mzroll/eicwidget.cpp:2312]: Okayy.. I will make the change

**Pull request merged**